lib/xenops_utils.ml

```diff
@@ -379,17 +379,24 @@ module FileFS = struct
 	let rm path =
 		List.iter
 			(fun path ->
-				if Sys.is_directory path then begin
-					try
-						debug "DB.delete %s" path;
-						Unix.rmdir path
-					with Unix.Unix_error(Unix.ENOTEMPTY, _, _) as e->
-						error "Failed to DB.delete %s : %s" path (Printexc.to_string e);
-						()
-				end else begin
-					debug "DB.delete %s" path;
-					Unix.unlink path;
-				end
+				debug "DB.delete %s" path;
+				try
+				  if Unix.((lstat path).st_kind = S_DIR)
+				  then Unix.rmdir path
+				  else Unix.unlink path;
```

Comment:
I would expect the following to be slightly more robust 
```
try Unix.rmdir path with _ -> () ;
try Unix.unlick path with _ -> ()
```

(however ideally `paths_of` would give you both the path and information about whether the path refers to a dir or a file)